shm/main.cpp

```diff
@@ -28,7 +27,11 @@ int main() {
     std::cout << "Speed: " << FD.getSpeed() << '\n';
     std::cout << "Name: " << FD.getName() << '\n';
     std::cout << "Id: " << FD.getId() << '\n';
-
+    
+    //Checking observer pattern
+    Time time;
+    Item knife("SilverKnife", 2, 30, 1, &time);
+    Ship::getName(Ship::getCargo(0));
```

Comment:
przez to się nie kompiluje, jeżeli chcesz wywołać getName albo getCargo to musisz wywołać na obiekcie, powyżej jest stworzony obiekt FD... nie wiem do końca co chcesz zrobić getName zwraca name statku, nie możesz tam dać argumentu
knife.getName() zwróci SilverKnife, itp.
może FD.getCargo(0) ?

albo żeby dodać cargo na statek najpierw trzeba sprawdzić czy taki obiekt istnieje (knife) - sprawdzi to funkcja findCargo jej wynik pójdzie do funkcji load. funkcja load jak dostanie iterator od findCargo różny od enda doda ilość noży do noży w ładowni statku, jeżeli findCargo nie znajdzie noży zwróci iterator end i funkcja load zrobi push_back do wektora cargo = FD.load(knife)